### PR TITLE
KOGITO-2811 Skip UI tests when running Maven with -DskipTests

### PR DESCRIPTION
**Pull request by cristianonicolai**
Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Comment by radtriste** [comment]
@cristianonicolai CI is failing due to 
```
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.9.1:yarn (yarn run build) on project kogito-apps-ui-packages: Failed to run task: 'yarn run build:prod --registry=https://repository.engineering.redhat.com/nexus/repository/registry.npmjs.org/' failed. org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1) -> [Help 1]
```

This is related to https://github.com/kiegroup/kogito-apps/blob/e628f89c9755eb583c65eab8f2eb9e56592e56e7/ui-packages/pom.xml#L154 as the plugin sets the registry as `--registry {REGISTRY_URL}` in the command execution `yarn run build:prod` and then, in the common package, you changed the line https://github.com/kiegroup/kogito-apps/pull/329/files#diff-f5260371af935f41092c381db01e865eR14 and the append of the registry is not supported by `yarn run lint` ... (got the same problem on Friday trying to deactivate the tests)

**Comment by radtriste** [comment]
this yarn bug is also failing the nightly: https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/blue/organizations/jenkins/KIE%2Fkogito%2Fkogito-runtimes-deploy/detail/kogito-runtimes-deploy/103/pipeline

**Comment by jstastny-cz** [comment]
I think we should separate the lint and webpack in this case.
It seems lint does not support the registry override - does it need a registry at all?
we can just add lint as a separate script in package.json and then invoke from maven using a separate frontend-maven-plugin execution without providing the registry.

For the sake of ease of local development, there can be multiple scripts in package.json - build:prod as was here up till now for ease of local development and then scripts doing things in isolation that could be invoked from the frontend-maven-plugin executions.

**Comment by jstastny-cz** [comment]
But I can't still see how we're actually reflecting the -DskipTests here ...

**Comment by cristianonicolai** [comment]
@jstastny-cz this does the trick https://github.com/kiegroup/kogito-apps/pull/329/files#diff-12fac6a8e5ae5fd5cb294aae9ba49ea5R166 as it is tied to test phase.

**Review by nmirasch — APPROVED** [review]

**Review by jstastny-cz — APPROVED** [review]

**Review by pefernan — APPROVED** [review]
+1

**Review by danielezonca — APPROVED** [review]
LGTM

File: ui-packages/package.json [hunk 1]
```diff
@@ -11,6 +11,8 @@
     "build": "lerna run build --stream --",
     "build:fast": "lerna run build:fast --stream --",
     "build:prod": "lerna run build:prod --stream --",
+    "test": "lerna run test --stream --",
+    "lint": "lerna run lint --stream",
```
Comment (jstastny-cz) [inline ui-packages/package.json]:
```suggestion
    "lint": "lerna run lint --stream --",
```

File: ui-packages/packages/management-console/package.json [hunk 2]
```diff
@@ -10,13 +10,12 @@
   },
   "scripts": {
     "precommit": "lint-staged",
-    "build:prod": "yarn run lint && webpack --config webpack.prod.js && yarn run test:coverage",
+    "build:prod": "yarn run lint && webpack --config webpack.prod.js",
```
Comment (radtriste) [inline ui-packages/packages/management-console/package.json:13]: Seems this line is creating some problems in CI:
https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/blue/rest/organizations/jenkins/pipelines/KIE/pipelines/kogito/pipelines/ghprb-webhooks/pipelines/kogito-apps/runs/647/nodes/74/steps/107/log/?start=0

```
[INFO] --- frontend-maven-plugin:1.9.1:yarn (yarn build) @ kogito-apps-ui-packages ---
[INFO] Running 'yarn run build:prod --registry=https://repository.engineering.redhat.com/nexus/repository/registry.npmjs.org/' in /home/jenkins/workspace/KIE/kogito/ghprb-webhooks/kogito-apps/kogito-apps/ui-packages
[INFO] yarn run v1.19.1
[INFO] $ lerna run build:prod --stream -- --registry=https://repository.engineering.redhat.com/nexus/repository/registry.npmjs.org/
[INFO] lerna notice cli v3.22.1
[INFO] lerna info ci enabled
[INFO] lerna info Executing command in 4 packages: "yarn run build:prod --registry=https://repository.engineering.redhat.com/nexus/repository/registry.npmjs.org/"
[INFO] @kogito-apps/common: $ yarn run lint && webpack --mode production --registry=https://repository.engineering.redhat.com/nexus/repository/registry.npmjs.org/
[INFO] @kogito-apps/common: $ tslint -c ./tslint.json --project . './src/**/*.ts{,x}'
[INFO] @kogito-apps/common: strict-type-predicates does not work without --strictNullChecks
[INFO] @kogito-apps/common: Insufficient number of arguments or no entry found.
[INFO] @kogito-apps/common: Alternatively, run 'webpack(-cli) --help' for usage info.
[INFO] @kogito-apps/common: Hash: a06bcb66bd52d67773cf
[INFO] @kogito-apps/common: Version: webpack 4.43.0
[INFO] @kogito-apps/common: Time: 38ms
[INFO] @kogito-apps/common: Built at: 07/21/2020 7:44:15 AM
[INFO] @kogito-apps/common: ERROR in Entry module not found: Error: Can't resolve './src' in '/home/jenkins/workspace/KIE/kogito/ghprb-webhooks/kogito-apps/kogito-apps/ui-packages/packages/common'
[INFO] @kogito-apps/common: error Command failed with exit code 2.
[INFO] @kogito-apps/common: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
[INFO] lerna ERR! yarn run build:prod --registry=https://repository.engineering.redhat.com/nexus/repository/registry.npmjs.org/ exited 2 in '@kogito-apps/common'
[INFO] error Command failed with exit code 2.
[INFO] info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by radtriste** [comment]
we should solve this problem asap as it is blocking examples from compiling because of new artifacts not deployed on maven snapshot repo

**Comment by jstastny-cz** [comment]
@cristianonicolai why did you revert latest changes? if `build:prod` does not make sense in common package, it should be enough just to remove the script and lerna run should take care of things.

**Comment by cristianonicolai** [comment]
@jstastny-cz @radtriste I need to wait for https://github.com/kiegroup/kogito-apps/pull/285 to be merged so I can actually apply a fix on top.

**Review by radtriste — APPROVED** [review]

**Comment by radtriste** [comment]
@cristianonicolai pipeline got through :)

**Pull request merged**